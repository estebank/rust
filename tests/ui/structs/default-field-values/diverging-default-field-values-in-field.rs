#![feature(default_field_values)]
#![allow(dead_code)]

mod a {
    pub struct Foo {
        pub something: i32 = 2,
    }

    pub struct Bar {
        pub foo: Foo = Foo { something: 10 }, //~ ERROR default field overrides that field's type's default
    }
}

mod b {
    pub enum Foo {
        X {
            something: i32 = 2,
        }
    }

    pub enum Bar {
        Y {
            foo: Foo = Foo::X { something: 10 }, //~ ERROR default field overrides that field's type's default
        }
    }
}

fn main() {
    let x = a::Bar { .. };
    let y = a::Foo { .. };
    assert_eq!(x.foo.something, y.something);
    let x = b::Bar::Y { .. };
    let y = b::Foo::X { .. };
    match (x, y) {
        (b::Bar::Y { foo: b::Foo::X { something: a} }, b::Foo::X { something:b }) if a == b=> {}
        _ => panic!(),
    }
}
