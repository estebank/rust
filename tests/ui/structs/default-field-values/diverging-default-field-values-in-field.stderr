error: default field overrides that field's type's default
  --> $DIR/diverging-default-field-values-in-field.rs:10:41
   |
LL |         pub foo: Foo = Foo { something: 10 },
   |                        ---              ^^ this overrides the default of field `something` in `Foo`
   |                        |
   |                        when constructing this value
   |
note: this field's default value in `Foo` is overriden
  --> $DIR/diverging-default-field-values-in-field.rs:6:30
   |
LL |     pub struct Foo {
   |     --------------
LL |         pub something: i32 = 2,
   |                              ^
   = note: `#[deny(default_field_overrides_default_field)]` on by default

error: default field overrides that field's type's default
  --> $DIR/diverging-default-field-values-in-field.rs:23:44
   |
LL |             foo: Foo = Foo::X { something: 10 },
   |                        ------              ^^ this overrides the default of field `something` in `X`
   |                        |
   |                        when constructing this value
   |
note: this field's default value in `X` is overriden
  --> $DIR/diverging-default-field-values-in-field.rs:17:30
   |
LL |         X {
   |         -
LL |             something: i32 = 2,
   |                              ^

error: aborting due to 2 previous errors

